lib/compiler/src/compile.erl

```diff
@@ -719,7 +719,9 @@ core_passes() ->
 
 kernel_passes() ->
     %% Optimizations that must be done after all other optimizations.
-    [{pass,sys_core_bsm},
+    [{pass,sys_core_alias},
```

Comment:
The optimization passes in `kernel_passes/0` are placed there for correctness reasons, that is, applying further optimizations would be unsafe.

If `sys_core_alias` does not introduce anything that would make further optimizations unsafe, it should be placed in `core_passes/0`. I don't have a strong opinion yet exactly where it should be placed, but I suspect it would make most sense to run it as the last "true" core pass.